event_types/event_types.py

```diff
@@ -1374,6 +1462,151 @@ def partition_event_types(
         return event_types
 
 
+def partition_event_types_offsetwise(
```

Comment:
To keep the export events function cleaner and generally the API unchanged I would have a "master" function `partition_event_types` and there call either `partition_event_types_energy` or `partition_event_types_energy_and_offset` based on the value of `offset_partition` which can be `False` by default.